dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcPQProp/HvdcPQProp.mo

```diff
@@ -0,0 +1,114 @@
+  Connectors.BPin modeU1 (value (start = modeU10)) "Boolean assessing the mode of the control of converter 1: true if U mode (here a proportional Q regulation), false if Q mode (fixed Q)";
+  Connectors.BPin modeU2 (value (start = modeU20)) "Boolean assessing the mode of the control of converter 2: true if U mode (here a proportional Q regulation), false if Q mode (fixed Q)";
+  Connectors.ImPin NQ1 "Signal to change the reactive power of converter 1 depending on the centralized voltage regulation";
+  Connectors.ImPin NQ2 "Signal to change the reactive power of converter 2 depending on the centralized voltage regulation";
+
+  parameter Boolean modeU10 "Start value of the boolean assessing the mode of the control of converter 1";
+  parameter Boolean modeU20 "Start value of the boolean assessing the mode of the control of converter 2";
+  parameter Real QPercent1 "Percentage of the coordinated reactive control that comes from converter 1";
+  parameter Real QPercent2 "Percentage of the coordinated reactive control that comes from converter 2";
+
+  parameter Types.ReactivePowerPu Q1MinPu  "Minimum reactive power in p.u (base SnRef) at terminal 1 (receptor convention)";
+  parameter Types.ReactivePowerPu Q1MaxPu  "Maximum reactive power in p.u (base SnRef) at terminal 1 (receptor convention)";
+  parameter Types.ReactivePowerPu Q2MinPu  "Minimum reactive power in p.u (base SnRef) at terminal 2 (receptor convention)";
+  parameter Types.ReactivePowerPu Q2MaxPu  "Maximum reactive power in p.u (base SnRef) at terminal 2 (receptor convention)";
+
+  Types.Angle Theta1(start = UPhase10) "Angle of the voltage at terminal 1 in rad";
+  Types.Angle Theta2(start = UPhase20) "Angle of the voltage at terminal 2 in rad";
+
+protected
+
+  parameter Types.Angle UPhase10 "Start value of voltage angle and filtered voltage angle at terminal 1 in rad";
+  parameter Types.Angle UPhase20 "Start value of voltage angle and filtered voltage angle at terminal 2 in rad";
+  Types.ReactivePowerPu Q1RawPu (start = s10Pu.im) "Reactive power of converter 1 without taking limits into account in p.u (base SnRef) (receptor convention)";
+  Types.ReactivePowerPu Q2RawPu (start = s20Pu.im) "Reactive power of converter 2 without taking limits into account in p.u (base SnRef) (receptor convention)";
+
+equation
+
+  Theta1 = Modelica.Math.atan2(terminal1.V.im,terminal1.V.re);
+  Theta2 = Modelica.Math.atan2(terminal2.V.im,terminal2.V.re);
+
+  Q1RawPu = Q1RefPu.value + QPercent1 * NQ1.value;
+  Q2RawPu = Q2RefPu.value + QPercent2 * NQ2.value;
+
+if running.value then
+
+// Reactive power regulation at terminal 1
+  if modeU1.value then
+    if Q1RawPu <= Q1MinPu then
+      Q1Pu = Q1MinPu;
+    elseif Q1RawPu >= Q1MaxPu then
+      Q1Pu = Q1MaxPu;
+    else
+      Q1Pu = Q1RawPu;
+    end if;
+  else
+    if Q1RefPu.value <= Q1MinPu then
+      Q1Pu = Q1MinPu;
+    elseif Q1RefPu.value >= Q1MaxPu then
+      Q1Pu = Q1MaxPu;
+    else
+      Q1Pu = Q1RefPu.value;
+    end if;
```

Comment:
I don't really like this if loop inside another if loop.
I don't find a completely nice way to rewrite it but something like:
Q1PuBla = if modeU1.value then Q1RawPu.value else Q1RefPu;
 if Q1PuBla <= QMinPu then
 a
elseif 
b
else
c

If it seems simpler for you, modify your code. Otherwise, keep it as it is.

Reply:
Ok, done